chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/summarystatistics/SummaryStatisticsCreatorJob.java

```diff
@@ -61,51 +74,71 @@ public SummaryStatisticsCreatorJob() throws Exception {
     }
 
     @Override
-    public void execute(final JobExecutionContext jobContext) throws JobExecutionException {
+    public void execute(JobExecutionContext jobContext) throws JobExecutionException {
         LOGGER.info("********* Starting the Summary Statistics Creation job. *********");
         try {
             SpringBeanAutowiringSupport.processInjectionBasedOnCurrentContext(this);
 
-            Boolean generateCsv = Boolean.valueOf(jobContext.getMergedJobDataMap().getString("generateCsvFile"));
-            Date startDate = getStartDate();
-            if (startDate == null) {
-                throw new RuntimeException("Could not obtain the startDate.");
-            }
-            Date endDate = new Date();
-            Integer numDaysInPeriod = Integer.valueOf(env.getProperty("summaryEmailPeriodInDays").toString());
+            LOGGER.info("Getting all listings.");
+            List<CertifiedProductDetailsDTO> allListings = certifiedProductDAO.findAll();
+            LOGGER.info("Completing getting all listings.");
 
-            EmailStatistics emailBodyStats = emailStatisticsCreator.getStatistics();
+            EmailStatistics emailBodyStats = emailStatisticsCreator.getStatistics(allListings);
+            saveSummaryStatistics(emailBodyStats);
 
+            Boolean generateCsv = Boolean.valueOf(jobContext.getMergedJobDataMap().getString("generateCsvFile"));
```

Comment:
Inside the `createSummaryStatisticsFile` method you use the private `isGenerateStatisticsFlagOn` method to immediately exit. Seems like this check only needs to happen in one place, and if it does happen in two they should both use this private method